scripts/hrrr_downloader.py

```diff
@@ -110,35 +172,54 @@ def maskDataset(ds: xr.Dataset, mask_file: str) -> xr.Dataset:
 def mergeDatasets(regionSubsetGribFiles: list) -> xr.Dataset:
   datasets = []
   dropVars = ["surface", "heightAboveGround", "valid_time", "step"]
-  # if f001, grab just the accumlated precip by dropping the other forecast variables
   dropVarsStep = dropVars + ["t", "r2", "si10", "sdswrf", "sdlwrf"]
+
   for f in regionSubsetGribFiles:
-    unMergedDatasets = cfgrib.open_datasets(f, indexpath="", decode_timedelta=False)
-    mergedDataset = xr.merge(
+    try:
+      groups = cfgrib.open_datasets(f, indexpath="", decode_timedelta=False)
+    except Exception as e:
+      print(f"cfgrib could not open {f}: {e} — skipping")
+      continue
+
+    # robust step detection
+    merged = xr.merge(
       [
         ds.drop_vars(dropVarsStep, errors="ignore")
-        if ds.step.values == np.timedelta64(1, "h")
+        if getattr(ds, "step", None) is not None
+        and np.array_equal(ds.step.values, np.timedelta64(1, "h"))
         else ds.drop_vars(dropVars, errors="ignore")
-        for ds in unMergedDatasets
+        for ds in groups
       ]
     )
-    mergedDataset.load()
-    datasets.append(mergedDataset)
+    try:
+      merged.load()
+    except Exception as e:
+      print(f"load() failed for {f}: {e} — skipping")
+      continue
+
+    datasets.append(merged)
+
+  if not datasets:
+    raise RuntimeError("No valid datasets to merge")
 
   other_vars = [ds for ds in datasets if "tp" not in ds.variables]
   tp_f001 = [ds for ds in datasets if "tp" in ds.variables]
 
-  tp_ds = xr.concat(tp_f001, dim="time")
-  other_ds = xr.concat(other_vars, dim="time")
-  combined_ds = xr.combine_by_coords([tp_ds, other_ds], compat="override")
-
-  # Wrap longitude values from 0-360 to -180 to 180
-  combined_ds["longitude"] = (combined_ds["longitude"] + 180) % 360 - 180
-  combined_ds["longitude"].attrs = {
-    "units": "degrees_east",
-    "standard_name": "longitude",
-    "long_name": "longitude",
-  }
+  parts = []
+  if tp_f001:
+    parts.append(xr.concat(tp_f001, dim="time"))
+  if other_vars:
+    parts.append(xr.concat(other_vars, dim="time"))
+  if not parts:
+    raise RuntimeError("No datasets after separating tp/other")
+
+  with xr.set_options(keep_attrs=True):
+    combined_ds = xr.combine_by_coords(parts, compat="override")
+
+  # convert lon to [-180, 180] if source was [0, 360]
+  if (combined_ds.longitude > 180).any():
+    combined_ds["longitude"] = combined_ds.longitude - 360
```

Comment:
This formula isn't correct. Please keep as it is on `main`

```suggestion
  if (combined_ds.longitude > 180).any():
    combined_ds["longitude"] = (combined_ds["longitude"] + 180) % 360 - 180
    combined_ds["longitude"].attrs = {
      "units": "degrees_east",
      "standard_name": "longitude",
      "long_name": "longitude",
  }
```

Reply:
Thanks for the suggestion! I have kept the formula as in main